lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -419,7 +419,7 @@ def update_tags(self, tags):
                 if not new_tags.get(key) or new_tags[key] != value:
                     changed = True
                     new_tags[key] = value
-            if isinstance(tags, dict):
+            if not self.append_tags and isinstance(tags, dict):
                 for key, value in tags.items():
                     if not self.module.params['tags'].get(key):
```

Comment:
I know this is not your changes, but this `update_tag` may return `Changed=True` when I want to add tags to a resource which has no tag. Would you mind add another conditional check to fix it?

Reply:
I suppose it's already the case. This is finely managed in lines above (418).

Reply:
the `module.params.get('tags') will be None if no tags here. Then break the rulesconditional check return False directly

Reply:
The `tags` argument corresponds to the current resource tags passed to the method and `module.params['tags']` are the new tags passed to the module. If the resource has no tag, `tags` will be `None` and the result dict `new_tags` will be set to `{}` before processing (line 415). If valid new tags are passed to the module (i.e. `module.params['tags']` is defined and is a dict), they'll be all added to `new_tags` and the method will return `True` (line 419).

I created a MWE, tested with the latest stable Ansible release (2.5.2), and with my fork also: https://gist.github.com/melmorabity/a7a7475b459b0611a236afc15f0ca7d4

Reply:
It is okay for me